src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -77,50 +82,134 @@ class CrossDeviceLink extends Component {
   }
 
   render = () =>
-    this.props.roomId ? <CrossDeviceLinkUI roomId={this.props.roomId} /> : <Spinner />
+    this.props.roomId ? <CrossDeviceLinkUI {...this.props}/> : <Spinner />
 }
 
 class CrossDeviceLinkUI extends Component {
   constructor(props) {
     super(props)
-    this.state = {copySuccess: false}
+    this.state = {
+      copySuccess: false,
+      sending: false,
```

Comment:
Consider moving copy and sending into their own components to make this component simpler

Reply:
Yes, I will definitely tidy up if there aren't issues with bigger priority to be fixed on this branch before tomorrow.